src/js/core/MesosMasterRequest.ts

```diff
@@ -0,0 +1,45 @@
+// @ts-ignore
+import * as MesosClient from "@dcos/mesos-client";
+import {
+  map,
+  retryWhen,
+  shareReplay,
+  catchError,
+  distinctUntilChanged
+} from "rxjs/operators";
+import { Observable, of } from "rxjs";
+
+import { linearBackoff } from "../utils/rxjsUtils";
+import Util from "#SRC/js/utils/Util";
+
+const { request } = MesosClient;
+const MAX_RETRIES = 5;
+const RETRY_DELAY = 500;
+
+export const MesosMasterRequestType = Symbol("MesosMasterRequest");
+
+export default request({ type: "GET_MASTER" }, "/mesos/api/v1?GET_MASTER").pipe(
```

Comment:
When is this every re-requested? If the master were to change while the UI was open, what would happen?